packages/@romejs/compiler/lint/rules/js/noUnusedVariables.test.ts

```diff
@@ -0,0 +1,49 @@
+import {test} from "rome";
+import {testLint} from "../testHelpers";
+import {dedent} from "@romejs/string-utils";
+
+test(
+	"enforce to not declare variables that are not used",
+	async (t) => {
+		await testLint(
+			t,
+			{
+				invalid: [
+					`const a = 4;`,
+					`let a = 4;`,
+					dedent`
+						function foo(myVar) {
+							console.log("foo");
+						};
+					`,
+					// dedent`
+					// 	function foo(myVar) {
+					// 		console.log("foo");
+					// 	};
+					// 	foo();
+					// `,
```

Comment:
Should be able to uncomment this now

```suggestion
					dedent`
						function foo(myVar) {
							console.log("foo");
						}
						foo();
					`,
```